src/app/pages/charts/line-chart-1/line-chart-1.component.ts

```diff
@@ -1,249 +1,57 @@
-import * as d3 from 'd3';
-import * as nv from 'nvd3';
+import { Component, ElementRef, OnInit } from '@angular/core';
 
-import { Component, OnInit } from '@angular/core';
+import { LineChartComponent } from 'theme/components/line-chart';
 
-// TODO: remove this
-const COLORS = {
-  red: '#f44336',
-  lightBlue: '#03a9f4',
-  orange: '#ffc107',
-  amber: '#ff9800',
-  teal: '#00bcd4',
-  purple: '#7726d3',
-  green: '#00d45a',
-  rowBgColor: '#4a4a4a',
-};
+import { LineChart1Service } from './line-chart-1.service';
 
 @Component({
   selector: 'app-line-chart-1',
-  styleUrls: ['./line-chart-1.component.scss'],
-  templateUrl: './line-chart-1.component.html',
+  styleUrls: ['../../../../theme/components/line-chart/line-chart.component.scss'],
+  template: ``,
+  providers: [LineChart1Service],
 })
-export class LineChart1Component implements OnInit {
-  private data;
-  private container;
-  private svg;
-  private svgHeight;
-  private svgWidth;
-  private barsLayout;
-  private lineChart;
-  private timer;
-  private maxX = 10;
-  private xStep = 0.125;
-  private xDrawStep = 4;
-  private rowBgColor = COLORS.rowBgColor;
-  private margin = 20;
-  private xAxis = 'X';
-  private yAxis = 'Y';
-  private animationTime = 400;
-  private columns = this.maxX / 2;
-  private drawStep = this.xStep * this.xDrawStep;
-  private durationResizeAnimation = 500;
+export class LineChart1Component extends LineChartComponent {
+  constructor(
+    el: ElementRef,
+    private lineChart1Service: LineChart1Service,
+  ) {
+    super(el);
 
-  public ngOnInit() {
-    this.container = d3.select('.line-chart-1__container');
-    this.data = [
+    this.xAxis = 'X';
+    this.yAxis = 'Y';
+    this.maxX = 10;
+
+    this.afterConfigure();
+
+    this.animatedData = [
       {
         values: [],
         key: 'Sin(x)',
-        color: COLORS.lightBlue,
```

Comment:
why do you decide to remove colors?

Reply:
There were not many entries in the file. I left this constant in some files.

I can make `Injectable` from this constant for whole app but I don't see necessity for this.